### PR TITLE
provide some more motivation

### PR DESCRIPTION
**Pull request by PeterSommerlad**
Sorry for only a tiny piece, but didn't have much time today. But I hope it helps

unfortunately, I first created a branch that I merged into my fork. Need some practice with forked repos.

It might be easier to give me write access to your repo, if you like.

### CODE REVIEW

**Comment by steve-downey** [comment]
I'll read through this in a bit. I'll also set up CODEOWNERS so I get notifications and also practice better hygiene on main. 

Looking at the examples in light of # your comment, yes, more context would help. Essentially these are the sorts of things I see written today to work around not having T& support. 

Read my last update on shallow const. If you are OK, where this behavior matches Boost, let me know, and I'll give you commit access and list you as a coauthor. 

Otherwise I'll just give you commit.

File: papers/optional-ref.md [hunk 1]
```diff
@@ -59,14 +61,16 @@ o.transform([&](auto c&){
 :::
 
 # Motivation
-Optionals holding references are common other than in the standard libary's implementation.  The desire for such a feature is well understood, and many optional types in commonly used libraries provide it, with the semanics proposed here.
+Optionals holding references are common other than in the standard libary's implementation (or even in one of the big open sources standard libraries).  The desire for such a feature is well understood, and many optional types in commonly used libraries provide it, with the semanics proposed here.
```
Comment (steve-downey) [inline papers/optional-ref.md]: I know that libc++ is trying to make sure of ABI stability. But they still `static_assert(!is_reference_v<value_type>` . Is anyone shipping a way for a user to write std::optional<T&> ?

File: papers/optional-ref.md [hunk 2]
```diff
@@ -59,14 +61,16 @@ o.transform([&](auto c&){
...
 
+In freestanding environments or for safety-critical libraries, an optional type over references is important to implement containers, that otherwise as the standard library either would cause undefined behavior when accessing an non-available element, throw an exception, or silently create the element. Returning a plain pointer for such an optional reference, as the core guidelines suggest, is a non-type-safe solution and doesn't protect in any way from accessing an non-existing element by a nullptr dereference. In addition, the monadic APIs of std::optional makes is especially attractive by streamlining client code receiving such an optional reference, in contrast to a pointer that requires an explicit nullptr check and de-reference.
```
Comment (steve-downey) [inline papers/optional-ref.md:77]: ☑️

**Review by steve-downey — CHANGES_REQUESTED** [review]

File: papers/optional-ref.md (hunk 1 above)
Comment (PeterSommerlad) [reply papers/optional-ref.md]: not to my knowledge. I was just referring to libc++. We can phrase it in a way to say that libc++ already ships an implementation that is disabled due to the current disallowance of optional<T&> by the standard. Or leave it out.

**Comment by PeterSommerlad** [comment]
amended as asked for.

File: papers/optional-ref.md (hunk 1 above)
Comment (PeterSommerlad) [reply papers/optional-ref.md]: see amended pull request. separate sentence on libc++.

**Review by steve-downey — APPROVED** [review]

**Pull request merged**